File: 3000.maximum-area-of-longest-diagonal-rectangle/solution.go
func areaOfMaxDiagonal(dimensions [][]int) int {
    a := 0
    d := 0
    for _, r := range dimensions {
        x, y := r[0], r[1]
        if h := x * x + y * y; h > d {
            d = h
            a = x * y
        } else if h == d && x * y > a {
            a = x * y
        }
    }
    return a
}

File: 3000.maximum-area-of-longest-diagonal-rectangle/solution.kt
class Solution {
    fun areaOfMaxDiagonal(dimensions: Array<IntArray>): Int {
        return dimensions
            .maxWith(
                compareBy(
                    { (l, w) -> l * l + w * w },
                    { (l, w) -> l * w }
                )
            )
            .let { (l, w) -> l * w }
    }
}